### PR TITLE
Fix wrong Sync state detection

### PR DESCRIPTION
**Pull request by illuque-iov**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/net/BlockSyncService.java [hunk 1]
```diff
@@ -150,7 +150,23 @@ private void tryReleaseStore(long bestBlockNumber) {
         }
     }
 
+    public boolean isReadyToAcceptBlocks() {
+        if (blockchain.getSize() == 1) { // empty blockchain
+            return false;
```
Comment (illuque-iov) [inline rskj-core/src/main/java/co/rsk/net/BlockSyncService.java:155]: I'm not sure about this check. Intuitively, I would say that an empty blockchain should be ready to accept blocks, but I've seen dedicated tests suggesting the contrary:
- `co.rsk.net.NodeBlockProcessorTest#noSyncingWithEmptyBlockchain`
- `co.rsk.net.AsyncNodeBlockProcessorTest#noSyncingWithEmptyBlockchain`

Comment (joaquinlpereyra) [reply rskj-core/src/main/java/co/rsk/net/BlockSyncService.java:155]: I have looked into this a bit. What I can say is that:
1. `isReadyToAcceptBlocks` refers specifically to a single type of message, `NewBlockHashMessage`, and probably the method should be renamed to reflect that.
2. Is possible, although surprising, that the intention behind the tests was to prevent blocks from receiving new block hashes they did not requests while they where not synced, similar to what is being checked now by checking the current known block number with zero. 

(2) should be confirmed. The code is a bit hard to follow. What I recommend is setting up a node and checking the logs. Writing down in detail the sync process wouldn't hurt either.

File: rskj-core/src/main/java/co/rsk/net/BlockSyncService.java [hunk 2]
```diff
@@ -150,7 +150,23 @@ private void tryReleaseStore(long bestBlockNumber) {
...
+        }
+
+        if (getLastKnownBlockNumber() == 0) { // recently restarted, no info about lastKnownBlock yet
```
Comment (joaquinlpereyra) [inline rskj-core/src/main/java/co/rsk/net/BlockSyncService.java:158]: This should return false here? We don't want new block hashes when we recently started, we are busy syncing.

File: rskj-core/src/test/java/co/rsk/core/SnapshotManagerTest.java [hunk 3]
```diff
@@ -133,7 +140,6 @@ public void revertToNonExistentSnapshot() {
     @Test
     public void revertToSnapshot() {
         addBlocks(10);
-
```
Comment (joaquinlpereyra) [inline rskj-core/src/test/java/co/rsk/core/SnapshotManagerTest.java:136]: More tests besides cosmetic changes would be nice :) also, documenting why some tests are there (specially the surprising ones found on this PR) would be great.

Comment (illuque-iov) [reply rskj-core/src/test/java/co/rsk/core/SnapshotManagerTest.java:136]: Cosmetic changes were due to the IDE auto-formatting, the only change was on line 379: `Assert.assertFalse(processor.isReadyToAcceptBlocks());`

This PR is preliminar to showcase the spotted scenarios, not production ready, so changes are expected and also more tests.

File: rskj-core/src/main/java/co/rsk/net/BlockSyncService.java (hunk 1 above)
Comment (illuque-iov) [reply rskj-core/src/main/java/co/rsk/net/BlockSyncService.java:155]: Thanks Joaco, will take a look.
